vaadin-testbench-unit-junit5/src/test/java/com/vaadin/flow/component/routerlink/RouterLinkTesterTest.java

```diff
@@ -0,0 +1,183 @@
+
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.router.RouteConfiguration;
+import com.vaadin.flow.router.RouterLink;
+import com.vaadin.testbench.unit.ComponentTester;
+import com.vaadin.testbench.unit.UIUnitTest;
+import com.vaadin.testbench.unit.ViewPackages;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+@ViewPackages
+class RouterLinkTesterTest extends UIUnitTest {
+
+    private ComponentTester<RouterLinkView> $routerLinkView;
+
+    @BeforeEach
+    void init() {
+        RouteConfiguration.forApplicationScope()
+                .setAnnotatedRoute(RouterLinkView.class);
+        RouteConfiguration.forApplicationScope()
+                .setAnnotatedRoute(RouterLinkStaticTargetView.class);
+        RouteConfiguration.forApplicationScope()
+                .setAnnotatedRoute(RouterLinkUrlParameterTargetView.class);
+        RouteConfiguration.forApplicationScope()
+                .setAnnotatedRoute(RouterLinkQueryParameterTargetView.class);
+        RouteConfiguration.forApplicationScope()
+                .setAnnotatedRoute(RouterLinkRouteParameterTargetView.class);
+
+        var routerLinkView = navigate(RouterLinkView.class);
+        $routerLinkView = test(routerLinkView);
+    }
+
+    @Test
+    void routerLink_targetless() {
+        // get router link
+        var targetlessRouterLink = $routerLinkView.find(RouterLink.class)
+                .withText("No Target")
+                .single();
+        var $targetlessRouterLink = test(targetlessRouterLink);
+        Assertions.assertNotNull($targetlessRouterLink,
+                "Tester for targetless RouterLink not initialized.");
+
+        // verify its href
+        Assertions.assertEquals("",
+                $targetlessRouterLink.getHref());
+
+        // verify its click action fails due to no navigation target
+        Assertions.assertThrows(IllegalStateException.class, $targetlessRouterLink::click);
+    }
+
+    @Test
+    void routerLink_static() {
+        // get router link
+        var staticRouterLink = $routerLinkView.find(RouterLink.class)
+                .withText("Static Target")
+                .single();
+        var $staticRouterLink = test(staticRouterLink);
+        Assertions.assertNotNull($staticRouterLink,
+                "Tester for static RouterLink not initialized.");
+
+        // verify its href
+        Assertions.assertEquals(RouterLinkStaticTargetView.ROUTE,
+                $staticRouterLink.getHref());
+
+        // verify its click action returns correct target
+        var targetView = $staticRouterLink.click();
+        Assertions.assertInstanceOf(RouterLinkStaticTargetView.class, targetView);
+
+        // verify navigation target is correct
+        var $targetView = test(targetView);
+        Assertions.assertDoesNotThrow(() -> $targetView.find(Span.class)
+                .withText("Static Target View")
+                .single());
```

Comment:
Same as for `RouterLink`, what about having the Span as a package protected field and simply assert on its `getText()`?